landscaper/gardenlet/pkg/gardenlet.go

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pkg
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+
+	v2 "github.com/gardener/component-spec/bindings-go/apis/v2"
+	"github.com/gardener/component-spec/bindings-go/apis/v2/cdutils"
+	"github.com/gardener/component-spec/bindings-go/codec"
+	landscaperv1alpha1 "github.com/gardener/landscaper/apis/core/v1alpha1"
+	landscaperconstants "github.com/gardener/landscaper/apis/deployer/container"
+	"github.com/sirupsen/logrus"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	confighelper "github.com/gardener/gardener/pkg/gardenlet/apis/config/helper"
+	configv1alpha1 "github.com/gardener/gardener/pkg/gardenlet/apis/config/v1alpha1"
+	"github.com/gardener/gardener/pkg/logger"
+
+	"github.com/gardener/gardener/landscaper/gardenlet/pkg/apis/imports"
+)
+
+const (
+	gardenerComponentName = "github.com/gardener/gardener"
+	gardenletImageName    = "gardenlet"
+)
+
+// Landscaper has all the context and parameters needed to run a Gardenlet landscaper.
+type Landscaper struct {
+	log                      *logrus.Entry
+	gardenClient             kubernetes.Interface
+	seedClient               kubernetes.Interface
+	// using internal version of the import configuration
+	Imports                  *imports.Imports
+	// working on the external version of the GardenletConfiguration file
+	gardenletConfiguration   *configv1alpha1.GardenletConfiguration
+	landscaperOperation      string
+	gardenletImageRepository string
+	gardenletImageTag        string
+
+	// disables certain checks that require Gardener API groups in the Garden cluster
+	isIntegrationTest bool
```

Comment:
```suggestion
	disableGardenChecks bool
```
I don't think that knowledge about "integration test" should be part of the API or implementation of any component. Instead, the component should provide (if appropriate, multiple) settings to enable / disable things that could be then used by integration tests, or for other purposes, if needed.

Reply:
Together with @schrodit  and @rfranzke we agreed to have a temporary and simple way to run the landscaper gardenelet for integration tests without requiring a full Gardener installation.
For me that's the most simple way and I am also not sure how to express that in the import configuration.

I would leave it like this as it is only temporary. But of course more opinions are welcome

Reply:
unresolve if necessary